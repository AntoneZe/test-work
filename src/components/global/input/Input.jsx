import React from "react";
import PropTypes from "prop-types";

import "./input.less";

const Input = ({ type, id }) => {
  return <input type={type} id={id} className="custom-input" />;
};

Input.propTypes = {
  type: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default Input;
