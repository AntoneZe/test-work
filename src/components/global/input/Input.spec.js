import React from "react";

import Input from "./Input";

const type = "text";
const id = "test-1";
const handler = () => "done";
const errorMessage = "test message";

describe("should render Input component", () => {
  it("should contain .custom-input__wrapper class", () => {
    let setUp = (props) => shallow(<Input {...props} />);
    const component = setUp({ type, id, handler });

    const input = component.find(".custom-input__wrapper");
    expect(input.length).toBe(1);
  });

  it("should contain error message", () => {
    let setUp = (props) => shallow(<Input {...props} />);
    const component = setUp({ type, id, handler, errorMessage });
    const input = component.find("span");
    expect(input.length).toBe(1);
  });
});

describe("Input component", () => {
  it("should render Input component with props", () => {
    const component = shallow(
      <Input
        type={"text"}
        id={"test-2"}
        handler={() => {
          return "done";
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});

describe("component callback", () => {
  it("should handle Button with props", () => {
    const mockCallBack = jest.fn();
    const event = {
      preventDefault() {},
      target: {
        value: "test",
      },
    };

    const component = shallow(
      <Input id={id} type={type} handler={mockCallBack} />
    );
    component.find("input").simulate("change", event);
    expect(mockCallBack).toBeCalledWith("test");
  });
});
