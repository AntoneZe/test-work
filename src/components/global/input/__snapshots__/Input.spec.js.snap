// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render Input component with props 1`] = `
<div
  className="custom-input__wrapper"
>
  <input
    className="custom-input"
    id="test-2"
    onChange={[Function]}
    type="text"
  />
</div>
`;
